mvvm/src/main/java/com/thefuntasty/mvvm/BaseViewModel.kt

```diff
@@ -117,4 +122,28 @@ abstract class BaseViewModel : ViewModel(), Observable, LifecycleObserver {
         }
         callbacks?.notifyCallbacks(this, fieldId, null)
     }
+
+    /**
+     * Simultaneous click protection. Ignores multi clicks, missclicks and other unwanted
+     * behaviour by ignoring code block in onDebounce DSL block for time period
+     * specified by clickDebounce param
+     * clickDebbounce default value is set to 1000L which represents 1 second delay.
+     * clickDebounce param is overridable
+     *
+     * @param block executes only when there is delay larger than clickDebounce
+     */
+    fun onDebounce(block: () -> Unit) {
```

Comment:
Takisto ak pouzijes viacej onDebounce metod v jednom ViewModeli tak kliknutie na jeden button ti zablokuje click na druhy

Reply:
to je prave wanted funcionality

Reply:
Nieco taketo by malo skor byt implementovane za pomoci disablovania a enablovanie buttonov ked uz to je potrebne

Reply:
Disablovat by si mohol aj iny button, to by zas ale kod bol viac komplexni... Asi nemam lepsie riesenie na to, ale tu sa mi taktiez nepaci to ze je to sucastou BaseViewModel co je celkovo zle, lebo ak sa zacnu taketo utility pridavat sem, tak velkost tejto triedy zacne narastat. Ked uz chceme aby to bolo implementovane takto tak to extrahuj do samostanej triedy ktoru mozes potom pridat tam kde potrebujes.